ui/components/failed.jsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import ImageDiff from 'react-image-diff';
+import {Clearfix, ButtonGroup, Button,
+  Jumbotron, Grid, Row, Col, Panel}
+  from 'react-bootstrap';
+
+var Failed = React.createClass({
+  getInitialState: function() {
+    return {
+      view: 'default',
+      isDefault: true,
+      is2Up: false,
+      value: 0.5,
+      openError: false
+    };
+  },
+  /* handler for swipe and fade value*/
+  changeValue: function(element) {
+    this.setState({
+      value: Number(element.target.value)
+    });
+  },
+  /* handler for switching between views */
+  changeView: function(element) {
+    var selectedView = element.target.name;
+    this.setState({
+      view: selectedView,
+      isDefault: selectedView === 'default' ? true : false,
+      is2Up: selectedView === '2-up' ? true : false
+    });
+  },
+  /* handler for the error text box */
+  openErrorMessage: function() {
+    this.setState({
+      openError: !this.state.openError
+    });
+  },
+  render: function() {
+    var _this = this,
+        baseline = this.props.paths.baseline,
+        screenshot = this.props.paths.screenshot,
+        diff = this.props.paths.diff,
+        error = this.props.error,
+        buttons = [],
+        report;
+
+    /* buttons to select the report type */
+    ['default', '2-up', 'swipe', 'fade'].forEach(function(item) {
```

Comment:
You can move this into [statics](https://facebook.github.io/react/docs/component-specs.html#statics) and access them with `this.constructor.ceva`